--- tabbed.c	2017-01-20 18:13:07.708069915 -0800
+++ tabbed.c.new	2017-01-20 18:12:59.738070342 -0800
@@ -736,7 +736,6 @@
 			        sizeof(Client *) * (nclients - nextpos - 1));
 
 		clients[nextpos] = c;
-		updatetitle(nextpos);
 
 		XLowerWindow(dpy, w);
 		XMapWindow(dpy, w);
@@ -757,10 +756,11 @@
 		/* Adjust sel before focus does set it to lastsel. */
 		if (sel >= nextpos)
 			sel++;
-		focus(nextfocus ? nextpos :
-		      sel < 0 ? 0 :
-		      sel);
-		nextfocus = foreground;
+		updatetitle(nextpos);
+		if (focusnew || nclients == 1) {
+			focus(nextfocus ? nextpos : sel < 0 ? 0 : sel);
+			nextfocus = foreground;
+		}
 	}
 }
 
@@ -1253,7 +1253,7 @@
 void
 usage(void)
 {
-	die("usage: %s [-dfksv] [-g geometry] [-n name] [-p [s+/-]pos]\n"
+	die("usage: %s [-defklsv] [-g geometry] [-n name] [-p [s+/-]pos]\n"
 	    "       [-r narg] [-o color] [-O color] [-t color] [-T color]\n"
 	    "       [-u color] [-U color] command...\n", argv0);
 }
@@ -1273,6 +1273,9 @@
 	case 'd':
 		detach = True;
 		break;
+	case 'e':
+		focusnew = True;
+		break;
 	case 'f':
 		fillagain = True;
 		break;
@@ -1282,6 +1285,9 @@
 	case 'k':
 		killclientsfirst = True;
 		break;
+	case 'l':
+		focusnew = False;
+		break;
 	case 'n':
 		wmname = EARGF(usage());
 		break;
